### ISSUE TITLE
Clicking on notification window crashes Mattermost when owned by root

### ISSUE THREAD

**Issue by ciakval** [mattermost/desktop#2131]
#### Summary
Clicking on the notification window crashes the Mattermost desktop app when its directory is not user-writable

#### Steps to reproduce

* Fedora Workstation 36
* mattermost-desktop-5.1.0-linux-x64.tar
* extract to */opt/mattermost-desktop-5.1.0-linux-x64/*
* generate the Desktop Launcher and moved to *$HOME/.local/share/applications/Mattermost.desktop*
* run `chown -R root:root /opt/mattermost-desktop-5.1.0-linux-x64/`
* Start Mattermost
* receive a notification
* click on it

#### Expected behavior
Click on the notification brings Mattermost window to the top and takes me to the channel where the notification came from.

#### Observed behavior (that appears unintentional) 
Mattermost application crashes and needs to be restarted.

#### Possible fixes
Don't change ownership of the unpacked application to *root* with the `chown` step

**Comment by devinbinnie** [comment]
I'm unable to reproduce this myself on Fedora 35. Is this consistently reproducible for you?
If so, can you turn on Debug logging in the Settings window and post the output here? There may also be an `uncaughtException` file that may contain more information as well. You can find both of these in: `~/.config/Mattermost`

**Comment by nab-77** [comment]
@ciakval Not wanting to hijack the moment but would you be interested in testing the new RPM/repo for the desktop we have coming soon?

**Comment by ciakval** [comment]
So I did some digging.

@devinbinnie it is not always reproducible, but most of the time yes. Do you know of any good way to trigger a notification for myself (I don't have a secondary account but maybe I could find an open server and create two accounts there?).

I have increased the log level to Debug and I found a coredump in `journalctl`, attached here.
[journal.log](https://github.com/mattermost/desktop/files/8785584/journal.log)

@nab-77 sure, we're using Mattermost at `$DAYJOB` so I can spend some (limited) time testing new RPMs.

**Comment by ciakval** [comment]
Reproduced when loglevel was set to debug, attaching the debug log and `journalctl` from the crash.
[crash-debug.log](https://github.com/mattermost/desktop/files/8785741/crash-debug.log)
[crash-journal.log](https://github.com/mattermost/desktop/files/8785743/crash-journal.log)

**Comment by devinbinnie** [comment]
So looks like it is crash when the notification is sent, but we don't have a stack trace. Was there an `uncaughtException` file in the aforementioned directory? That would contain a stack trace.

**Comment by ciakval** [comment]
I didn't find it (but had done a reboot in between). Since then, I could not reproduce the issue. If I manage to reproduce it, I will update the information.

**Comment by ciakval** [comment]
After I updated my system, I am no longer able to reproduce the issue. I guess it can be closed.

The updates that were done:

```
$ dnf history info 64
Transaction ID : 64
...
Packages Altered:
    Upgrade  btrfs-progs-5.18-1.fc36.x86_64                  @updates
    Upgraded btrfs-progs-5.16.2-1.fc36.x86_64                @@System
    Upgrade  gnome-keyring-42.1-1.fc36.x86_64                @updates
    Upgraded gnome-keyring-40.0-4.fc36.x86_64                @@System
    Upgrade  gnome-keyring-pam-42.1-1.fc36.x86_64            @updates
    Upgraded gnome-keyring-pam-40.0-4.fc36.x86_64            @@System
    Upgrade  hicolor-icon-theme-0.17-13.fc36.noarch          @updates
    Upgraded hicolor-icon-theme-0.17-12.fc36.noarch          @@System
    Upgrade  system-config-printer-libs-1.5.16-1.fc36.noarch @updates
    Upgraded system-config-printer-libs-1.5.15-6.fc36.noarch @@System
    Upgrade  system-config-printer-udev-1.5.16-1.fc36.x86_64 @updates
    Upgraded system-config-printer-udev-1.5.15-6.fc36.x86_64 @@System
    Upgrade  systemd-250.6-1.fc36.x86_64                     @updates
    Upgraded systemd-250.3-8.fc36.x86_64                     @@System
    Upgrade  systemd-container-250.6-1.fc36.x86_64           @updates
    Upgraded systemd-container-250.3-8.fc36.x86_64           @@System
    Upgrade  systemd-libs-250.6-1.fc36.x86_64                @updates
    Upgraded systemd-libs-250.3-8.fc36.x86_64                @@System
    Upgrade  systemd-networkd-250.6-1.fc36.x86_64            @updates
    Upgraded systemd-networkd-250.3-8.fc36.x86_64            @@System
    Upgrade  systemd-oomd-defaults-250.6-1.fc36.noarch       @updates
    Upgraded systemd-oomd-defaults-250.3-8.fc36.noarch       @@System
    Upgrade  systemd-pam-250.6-1.fc36.x86_64                 @updates
    Upgraded systemd-pam-250.3-8.fc36.x86_64                 @@System
    Upgrade  systemd-resolved-250.6-1.fc36.x86_64            @updates
    Upgraded systemd-resolved-250.3-8.fc36.x86_64            @@System
    Upgrade  systemd-udev-250.6-1.fc36.x86_64                @updates
    Upgraded systemd-udev-250.3-8.fc36.x86_64                @@System
    Upgrade  xorg-x11-server-Xwayland-22.1.2-1.fc36.x86_64   @updates
    Upgraded xorg-x11-server-Xwayland-22.1.1-1.fc36.x86_64   @@System
    
$ dnf history info 65
Transaction ID : 65
...
Packages Altered:
    Install  kernel-5.17.11-300.fc36.x86_64               @updates
    Install  kernel-core-5.17.11-300.fc36.x86_64          @updates
    Install  kernel-modules-5.17.11-300.fc36.x86_64       @updates
    Install  kernel-modules-extra-5.17.11-300.fc36.x86_64 @updates
    Upgrade  ansible-5.8.0-1.fc36.noarch                  @updates
    Upgrade  criu-3.17-2.fc36.x86_64                      @updates
    Upgrade  criu-libs-3.17-2.fc36.x86_64                 @updates
    Upgrade  kernel-headers-5.17.11-300.fc36.x86_64       @updates
    Upgrade  libbytesize-2.7-1.fc36.x86_64                @updates
    Upgrade  libnl3-3.6.0-3.fc36.x86_64                   @updates
    Upgrade  libnl3-cli-3.6.0-3.fc36.x86_64               @updates
    Upgrade  libnotify-0.7.12-1.fc36.x86_64               @updates
    Upgrade  librados2-2:16.2.9-1.fc36.x86_64             @updates
    Upgrade  librbd1-2:16.2.9-1.fc36.x86_64               @updates
    Upgrade  libxmlb-0.3.9-1.fc36.x86_64                  @updates
    Upgrade  logrotate-3.20.1-1.fc36.x86_64               @updates
    Upgrade  perl-libwww-perl-6.66-1.fc36.noarch          @updates
    Upgrade  python3-bytesize-2.7-1.fc36.x86_64           @updates
    Upgrade  selinux-policy-36.9-1.fc36.noarch            @updates
    Upgrade  selinux-policy-targeted-36.9-1.fc36.noarch   @updates
    Upgraded kernel-5.17.7-300.fc36.x86_64                @@System
    Upgraded kernel-core-5.17.7-300.fc36.x86_64           @@System
    Upgraded kernel-modules-5.17.7-300.fc36.x86_64        @@System
    Upgraded kernel-modules-extra-5.17.7-300.fc36.x86_64  @@System
    Upgraded ansible-5.7.0-1.fc36.noarch                  @@System
    Upgraded criu-3.17-1.fc36.x86_64                      @@System
    Upgraded criu-libs-3.17-1.fc36.x86_64                 @@System
    Upgraded kernel-headers-5.17.6-300.fc36.x86_64        @@System
    Upgraded libbytesize-2.6-3.fc36.x86_64                @@System
    Upgraded libnl3-3.5.0-9.fc36.x86_64                   @@System
    Upgraded libnl3-cli-3.5.0-9.fc36.x86_64               @@System
    Upgraded libnotify-0.7.11-1.fc36.x86_64               @@System
    Upgraded librados2-2:16.2.7-11.fc36.x86_64            @@System
    Upgraded librbd1-2:16.2.7-11.fc36.x86_64              @@System
    Upgraded libxmlb-0.3.7-1.fc36.x86_64                  @@System
    Upgraded logrotate-3.19.0-2.fc36.x86_64               @@System
    Upgraded perl-libwww-perl-6.65-1.fc36.noarch          @@System
    Upgraded python3-bytesize-2.6-3.fc36.x86_64           @@System
    Upgraded selinux-policy-36.8-2.fc36.noarch            @@System
    Upgraded selinux-policy-targeted-36.8-2.fc36.noarch   @@System

```
